tests/auth/src/lib.rs

```diff
@@ -482,4 +482,133 @@ mod test_b {
             }
         }
     }
+
+    mod wasm {
```

Comment:
In the end these tests are not really required or overly related to the fix. But I'd like to keep them because the test vectors of an authorising or declining auth are handy, and these tests test that there are no observable differences between auth against natively loaded check auth vs wasm loaded, which is test coverage we don't have today.

Reply:
Moved the tests to:
- #1334